test/unit/high-security.test.js

```diff
@@ -41,238 +35,278 @@ function getPath(obj, path) {
   return obj[paths[0]]
 }
 
-describe('high security mode', function () {
-  describe('config to be sent during connect', function () {
+tap.Test.prototype.addAssert('check', 3, function (key, before, after) {
+  var fromFile = { high_security: true }
+  setPath(fromFile, key, before)
+
+  var config = new Config(fromFile)
+  return this.same(getPath(config, key), after)
+})
+
+tap.Test.prototype.addAssert('checkServer', 4, function (config, key, expected, server) {
+  setPath(config, key, expected)
+  var fromServer = { high_security: true }
+  fromServer[key] = server
+
+  this.same(getPath(config, key), expected)
+  this.same(fromServer[key], server)
+
+  config.onConnect(fromServer)
+  this.same(getPath(config, key), expected)
+  return this.pass()
```

Comment:
I think you want to return the line above or just return nothing.  The tap docs are confusing because I did confirm assertions run just fine in a custom `addAssert` without returning.  If you return like this you get an empty assertion

```
  # Subtest: should not change transaction_events settings
                ok 1 - should be equivalent
                ok 2 - should be equivalent
                ok 3 - should be equivalent
                ok 4 - (unnamed test)
```

Reply:
Woof, fair point. I'll make the change 👍